language/move-vm/runtime/src/session.rs

```diff
@@ -187,19 +188,45 @@ impl<'r, 'l, S: MoveResolver> Session<'r, 'l, S> {
         sender: AccountAddress,
         gas_meter: &mut impl GasMeter,
     ) -> VMResult<()> {
-        self.runtime
-            .publish_module_bundle(modules, sender, &mut self.data_cache, gas_meter, true)
+        self.runtime.publish_module_bundle(
+            modules,
+            sender,
+            &mut self.data_cache,
+            gas_meter,
+            CompatibilityConfig::full_check(),
+        )
+    }
+
+    /// Same like `publish_module_bundle` but with a custom compatibility check.
+    pub fn publish_module_bundle_with_compat_config(
+        &mut self,
+        modules: Vec<Vec<u8>>,
+        sender: AccountAddress,
+        gas_meter: &mut impl GasMeter,
+        compat_config: CompatibilityConfig,
```

Comment:
I think we can migrate the `treat_friend_as_private` from `VerifierConfig` to `CompatibilityConfig`. Because the compat_config only affects the module publish. 
And the compat_config as argument of the publishing module function can be tested more easily.

Reply:
I try to refactor this https://github.com/move-language/move/pull/507. @vgao1996